### PR TITLE
1254 Added the changes for common automation framework to accept semvar

### PR DESCRIPTION
**Pull request by saathvika8**

### CODE REVIEW

File: manifests/containers/aws/build_agents/shared_service/terraform/manifest/manifest.xml
Comment (cmf-nttdata) [inline manifests/containers/aws/build_agents/shared_service/terraform/manifest/manifest.xml:1]: I think you're missing another entry at line 38.

File: manifests/python/manifest/manifest.xml
Comment (cmf-nttdata) [inline manifests/python/manifest/manifest.xml:1]: Missing another entry on line 9.

File: manifests/containers/aws/build_agents/shared_service/terraform/manifest/manifest.xml [hunk 1]
```diff
@@ -3,9 +3,9 @@
     This manifest defines the behavior of container components
 -->
 <manifest>
-  <project name="caf-components-container" path="components/container" remote="launch-dso-platform">
+  <project name="caf-components-container" path="components/container" remote="launch-dso-platform" revision="refs/tags/0.1.0" dso_override_attribute_revision='${CONTAINER_VER}'>
```
Comment (cmf-nttdata) [inline manifests/containers/aws/build_agents/shared_service/terraform/manifest/manifest.xml:6]: If we are going to add our own custom properties, the repo tool suggests that we begin the name with x-* as this is reserved for custom names. See here - https://gerrit.googlesource.com/git-repo/+/HEAD/docs/manifest-format.md#Element-project

I also think maybe the name is a little long? Maybe something like `x-override-revision` ?

Also, I'm not sure this even needed yet? Are we doing anything with these variables?

**Review by cmf-nttdata — CHANGES_REQUESTED** [review]

File: manifests/containers/aws/build_agents/shared_service/terraform/manifest/manifest.xml (hunk 1 above)
Comment (chris-taylor-ntt) [reply manifests/containers/aws/build_agents/shared_service/terraform/manifest/manifest.xml:6]: Looks like this was a customization we did to the git-repo tool:

https://github.com/nexient-llc/git-repo/blob/main/subcmds/envsubst.py#L134-L146

Have we tested this behavior with these replacement vars set to empty strings, or using `unset` to remove them entirely? I'm concerned that this has the possibility of overlaying a good value with an empty string.

File: templates/seed_makefile [hunk 2]
```diff
@@ -13,7 +13,7 @@
 # this will always point here
 REPO_MANIFESTS_URL ?= https://github.com/nexient-llc/common-automation-framework.git
 # this eventually will point to a git tag
-REPO_BRANCH ?= update
+REPO_MANIFESTS_VER ?= update
```
Comment (chris-taylor-ntt) [inline templates/seed_makefile:16]: This doesn't appear to be a valid branch in that repository.

File: manifests/containers/aws/build_agents/shared_service/terraform/manifest/manifest.xml (hunk 1 above)
Comment (saathvika8) [reply manifests/containers/aws/build_agents/shared_service/terraform/manifest/manifest.xml:6]: Yeah Chris, I have tested it without and with providing the dso_override_attribute_revision and if the value is not provided it simply keep the revision and if we want to update it to the override revision value , adding it to the Makefile will update it.

**Review by cmf-nttdata — APPROVED** [review]

**Pull request merged**